libafl/src/bolts/mod.rs

```diff
@@ -216,67 +216,55 @@ pub fn format_duration_hms(duration: &time::Duration) -> String {
     format!("{}h-{}m-{}s", (secs / 60) / 60, (secs / 60) % 60, secs % 60)
 }
 
+#[derive(Debug, Copy, Clone)]
+#[cfg(feature = "std")]
+enum Target {
+    Stdout,
+    Stderr,
+}
+
 /// A simple logger struct that logs to stderr when used with [`log::set_logger`].
 #[derive(Debug)]
 #[cfg(feature = "std")]
-pub struct SimpleStdErrLogger {
-    /// The min log level for which this logger will write messages.
-    pub log_level: Level,
+pub struct SimpleLogger {
+    /// Where the log goes
+    target: Target,
 }
 
 #[cfg(feature = "std")]
-impl SimpleStdErrLogger {
-    /// Create a new [`log::Log`] logger that will log [`Level::Trace`] and above
-    #[must_use]
-    pub const fn trace() -> Self {
-        Self {
-            log_level: Level::Trace,
-        }
-    }
-
-    /// Create a new [`log::Log`] logger that will log [`Level::Debug`] and above
-    #[must_use]
-    pub const fn debug() -> Self {
-        Self {
-            log_level: Level::Debug,
-        }
-    }
-
-    /// Create a new [`log::Log`] logger that will log [`Level::Info`] and above
-    #[must_use]
-    pub const fn info() -> Self {
-        Self {
-            log_level: Level::Info,
-        }
-    }
-
-    /// Create a new [`log::Log`] logger that will log [`Level::Warn`] and above
+impl SimpleLogger {
+    /// Create a new [`log::Log`] logger that will wrte log to stdout
     #[must_use]
-    pub const fn warn() -> Self {
+    pub const fn stdout() -> Self {
```

Comment:
I would also add a static method to register the loggers directly, like `SimpleLogger::register_stdout(LogLevel)` - then using it is super easy

Reply:
ok

Reply:
this needs to make it like
`SimpleLogger::register_logger(logger: &'static Self)`
because log::set_logger(X) needs X to be static
it sounds odd

Reply:
No, why? It can just use the static variable from the lib

Reply:
ok made two static logger strcut